### DISCUSSION TITLE
Raylib multiplayer?

### DISCUSSION THREAD

**Discussion by k-preslav** [raysan5/raylib#2598] [Q&A]
I want to make a multiplayer game with raylib but all i can find is a GitHub issue from 3 years ago where they say about rnet but i cant find anywhere to download it and there is no documentation about it. 

How can I make a multiplayer game in raylib???

**Comment by raysan5** [comment]
raylib does not support networking. I recommend using another library like [nbnet](https://github.com/nathhB/nbnet).

**Reply by k-preslav** [reply]
Wasn't rnet developed at some point? What happened with it?

**Reply by raysan5** [reply]
that project was abbandoned some time ago. It was broken and nobody was ready to work on it. That's what happens with open source some times...

**Reply by k-preslav** [reply]
Im really sorry for that 🥲

**Comment by kevyonan** [comment]
Networking is a huge kettle of fish.
Different games require different kinds of network topology and network design & architecture.

If raylib was to add a form of networking, it'd have to be very flexible and bare-bones in order to provide developers ways to make different network designs.

**Reply by anstropleuton** [reply]
Not really, raylib can choose to not make very flexible design because it wouldn't be the point of raylib
raylib can have a basic (but powerful) networking features and user can decide to use some other library
Anyways, good point

**Comment by Clement-Fernandes** [comment]
Hello!

With friends we made a game in `C++` that brings together a graphical library with **Raylib** and some **Networking** to play from 1 to 4 players on different computers!

The code of the repo is here: https://github.com/Studio-17/R-Type

There is documentation with `Doxygen` in each file if you want to understand the code much easily. :grin:

**Reply by raysan5** [reply]
@Clement-Fernandes Wow! Very nice! It looks like a very complete project! Do you have some gameplay video/trailer?

**Reply by Clement-Fernandes** [reply]
@raysan5 Sorry I haven't made one but I'll probably make a video of it in the coming week!

I have made another game with **Raylib** using **Networking** (not as complex as the other one because we didn't stay on it very long) but still pretty fun!

The code of the repo is here: https://github.com/Studio-17/Zappy

I'll probably make a video for that too. :laughing:
